src/useScript.ts

```diff
@@ -0,0 +1,164 @@
+import {useEffect, useState} from 'react';
+
+export interface LoadScriptOptions {
+  id?: string;
+  type?: string;
+  async?: boolean;
+  onload?: Function;
+  onerror?: Function;
+  onabort?: Function;
+  crossOrigin?: string | null;
+}
+
+export function makeScript(url, options : LoadScriptOptions) {
+  let element : HTMLScriptElement | null;
+  let _ready = false;
+  let _failed = false;
+  const {
+    type = "text/javascript",
+    async = true,
+    id,
+    onload,
+    onerror,
+    onabort,
+    crossOrigin = null
+  } = options || {};
```

Comment:
`charset`, `integrity`, `nonce`, `nomodule`, `referrerpolicy` and `defer`

the properties not exported.

see https://mdn.io/script-tag